src/Compilers/CSharp/Portable/Errors/DiagnosticInfoWithSymbols.cs

```diff
@@ -20,5 +21,17 @@ internal DiagnosticInfoWithSymbols(bool isWarningAsError, ErrorCode errorCode, o
         {
             this.Symbols = symbols;
         }
+
+        public override bool Equals(object obj)
+        {
+            return obj is DiagnosticInfoWithSymbols diws && 
+                   this.Symbols.Equals(diws.Symbols) && 
+                   base.Equals(obj);
+        }
```

Comment:
Are we handling `new DiagnosticInfo().Equals(new DiagnosticInfoWithSymbols())` as well?